src/python/pants/core/util_rules/external_tool.py

```diff
@@ -59,6 +59,22 @@ class DownloadedExternalTool:
     exe: str
 
 
+@dataclass(frozen=True)
+class ExternalToolVersion:
+    version: str
+    platform: str
+    sha256: str
+    filesize: int
+
+    def encode(self) -> str:
+        return "|".join([self.version, self.platform, self.sha256, str(self.filesize)])
+
+    @classmethod
+    def decode(cls, version_str: str) -> ExternalToolVersion:
+        version, platform, sha256, filesize = [x.strip() for x in version_str.split("|")]
```

Comment:
In the original `split_known_version_str` this expression was an generator-comprehension rather than a list-comprehension.

Probably not a big deal, but would prefer to keep it the same.

Reply:
I'm not sure a generator would work properly with unpacking. I think this is fine.

Reply:
That was how it was used previously. But no biggie for sure ;)

Reply:
I just tested it out, and I'm a bit surprised too, but looks like a genexpr works. The semantics are a bit different, though. The tuple unpack works (`a,s,d = (x for x in range(0,3))`), which is what we use. But `return (x for x in range(0,3))` returns the generator. It can be unpacked at the call site. But any uses expecting the auto-tuple-packing will be surprised.

Summary, either works, and I learned that you can unpack a generator.